backend/test_observer/controllers/artefacts/logic.py

```diff
@@ -89,6 +93,23 @@ def construct_dto_builds(
         yield build_dto
 
 
+def _derive_test_results(
```

Comment:
perhaps to avoid repeating some of the logic in `_derive_test_execution_status` we can instead have one function return both `status` and `test_results` as a tuple. We need a nice name for it though

Reply:
We can do this, I left it in two separate ones to simplify it, but we can combine it into something like: `_derive_test_execution_details` which would fetch both